### PR TITLE
First commit: build module + move GeneratorStartup extension

### PR DESCRIPTION
**Pull request by miovd**
Signed-off-by: RALAMBOTIANA MIORA <miora.ralambotiana@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
First commit

**What is the current behavior?** *(You can also link to an open issue here)*
The repository is empty

**What is the new behavior (if this is a feature change)?**
`GeneratorStartup` has been moved in this repository

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/network_hypothesis/GeneratorStartupAdderImpl.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.network_hypothesis;
```
Comment (geofjamg) [inline src/main/java/com/powsybl/network_hypothesis/GeneratorStartupAdderImpl.java]: Why not com.powsybl.network.hypothesis? I think we should avoid the _.

Comment (miovd) [reply src/main/java/com/powsybl/network_hypothesis/GeneratorStartupAdderImpl.java]: I followed the convention of `com.powsybl.balances_adjustment` but I agree, I prefer `com.powsybl.network.hypothesis` too

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        if: matrix.os != 'ubuntu-latest'
+        run: mvn --batch-mode install
+
+      - name: Run SonarCloud analysis
+        if: matrix.os == 'ubuntu-latest'
+        run: >
+          mvn --batch-mode -DskipTests sonar:sonar
+          -Dsonar.host.url=https://sonarcloud.io
+          -Dsonar.organization=powsybl-ci-github
+          -Dsonar.projectKey=com.powsybl:powsybl-core
```
Comment (geofjamg) [inline .github/workflows/maven.yml]: To fix

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,126 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (geofjamg) [inline pom.xml:1]: Licence header is missing

**Review by geofjamg — CHANGES_REQUESTED** [review]

**Review by geofjamg — APPROVED** [review]

File: src/main/java/com/powsybl/network/hypothesis/GeneratorStartup.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+package com.powsybl.network.hypothesis;
+
+import com.powsybl.commons.extensions.Extension;
+import com.powsybl.iidm.network.Generator;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public interface GeneratorStartup extends Extension<Generator> {
+
```
Comment (annetill) [inline src/main/java/com/powsybl/network/hypothesis/GeneratorStartup.java:16]: It seems that we are missing the start up cost of the generator, when its `P` is equal to zero.

Comment (miovd) [reply src/main/java/com/powsybl/network/hypothesis/GeneratorStartup.java:16]: I added a field to get this information, is it okay for you?

**Review by annetill — APPROVED** [review]
@geofjamg do you agree with Miora's adding?

**Comment by geofjamg** [comment]
> @geofjamg do you agree with Miora's adding?

yes

**Pull request merged**